app/packs/src/apps/mydb/elements/details/reactions/analysesTab/ReactionDetailsContainers.js

```diff
@@ -56,10 +58,29 @@ export default class ReactionDetailsContainers extends Component {
     this.handleOnClickRemove = this.handleOnClickRemove.bind(this);
     this.handleAccordionOpen = this.handleAccordionOpen.bind(this);
     this.handleSpChange = this.handleSpChange.bind(this);
+    this.onUIStoreChange = this.onUIStoreChange.bind(this);
   }
 
   componentDidMount() {
+    const { activeContainer } = this.state;
+    UIStore.listen(this.onUIStoreChange);
     TextTemplateActions.fetchTextTemplates('reaction');
+    if (this.containerRefs[activeContainer]) {
+      this.containerRefs[activeContainer].scrollIntoView({ behavior: 'smooth', block: 'start' });
+    }
+  }
+
+  componentWillUnmount() {
+    UIStore.unlisten(this.onUIStoreChange);
+  }
+
+  onUIStoreChange(state) {
+    const { activeContainer } = this.state;
+    if (state.reaction.activeContainer !== activeContainer) {
```

Comment:
If its not equal `state.reaction.activeContainer` is assigned to `this.state.activeContaiener`.
Couldn't you get rid of the `if` then? If it is equal, the assignment would not change anything, if I am not mistaken.

Reply:
To be honest, I mindlessly copied this from https://github.com/ComPlat/chemotion_ELN/blob/5ff46e0bbde4871d2f717fa763c3116402501f74/app/packs/src/apps/mydb/elements/details/samples/analysesTab/SampleDetailsContainers.js#L56.

It's probably meant to avoid calling `setState` unnecessarily. Now the question is, what's more expensive, the comparison or calling the setter? I'm guessing that calling the setter is more expensive, in which case I'd leave it as it is?

Reply:
Fair enough. Let's keep it like it is. Can `setState` also trigger a re-rendering? In this case, it should also not be called unnecessarily.